### PR TITLE
Add userPreferences Resource.

### PR DESCRIPTION
**Pull request by hibooboo2**
This resource is to allow accounts to have a list preferences with name and value where name is unique per account.

### CODE REVIEW

**Comment by sonchang** [comment]
LGTM

File: resources/content/schema/base/userPreference.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "resourceFields": {
+    "name": {
```
Comment (alena1108) [inline resources/content/schema/base/userPreference.json]: is the name supposed to be unique across accounts? or on per account basis?

Comment (hibooboo2) [inline resources/content/schema/base/userPreference.json]: per account.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**